src/controllers/mainController.ts

```diff
@@ -374,6 +383,27 @@ export default class MainController implements vscode.Disposable {
             vscode.commands.registerCommand(
             Constants.cmdScriptAlter, async (node: TreeNodeInfo) =>
             await this.scriptNode(node, ScriptOperation.Alter)));
+
+        // Copy object name keybinding
+        this._context.subscriptions.push(
+            vscode.commands.registerCommand(Constants.cmdCopyObjectName, async () => {
+                let node = this._objectExplorerProvider.currentNode;
+                // Folder node
+                if (node.contextValue === Constants.folderLabel) {
+                    return;
+                } else if (node.contextValue === Constants.serverLabel ||
+                    node.contextValue === Constants.disconnectedServerLabel) {
+                        await this._vscodeWrapper.clipboardWriteText(node.label);
```

Comment:
This is because the top most server can have a profile name which the user intends to copy, instead of the server name

Reply:
What makes you think that? Why would server be special in this regard? Other objects can have extra text added to the label too - why wouldn't we copy those in that case?

I'm not saying this is necessarily wrong - but I think it should be discussed before we just assume this is what customers want (PM involvement here would be good). Currently SSMS doesn't support copying the name of a server so there's no prior implementation to go off of here.

Reply:
```Other objects can have extra text added to the label too - why wouldn't we copy those in that case?```
This is incorrect as I stated in the above comment in the last sentence. We can remove copying the server name altogether, but copying the server name makes sense to me.

Reply:
Incorrect, we add text for some special types (such as system-versioned tables).

![image](https://user-images.githubusercontent.com/28519865/75276884-d7d85000-57bb-11ea-8208-1b19c0a6d4c9.png)

Reply:
This does have a `System-Versioned` tag in it, but it will never be copied (even by SSMS implementation). The point is you give a name to `Servers` and `Connections` and thus might want to copy the label it shows on the OE as it appears.

 Anyway, I've asked @yualan for his input in server labels/name copying experience.

Reply:
Yeah - I 100% agree that the behavior for the objects should just copy the name (and optional db/schema).

I just think for the server we should consider copying just the server name/port. Basically what you'd enter into a connection string to connect to that server. That's the use case I see being most common for that - but that's just my gut feeling so go with whatever Alan suggests there.